CIScripts/Report/GenerateTestReport.Tests.ps1

```diff
@@ -7,70 +7,126 @@ Describe "Generating test report" {
         ([Xml] $InputData).OuterXml
     }
 
-    BeforeAll {
+    function New-DummyFile {
+        Param([Parameter(Mandatory = $true)] [string] $Path)
+        '<test-results failures="0" inconclusive="0" skipped="0" date="2018-01-01" time="15:00:00">
+            <test-suite name="outer_suite" type="TestFixture" result="Success">
+                <results>
+                    <test-suite name="inner_suite" type="TestFixture" result="Success">
+                        <results>
+                            <test-case name="test" result="Success" />
+                        </results>
+                    </test-suite>
+                </results>
+            </test-suite>
+        </test-results>
+        ' | Set-Content -Path $Path
+    }
+
+    function New-TemporaryDirs {
         $InputDir = Join-Path $TestDrive "testReportInput"
         $OutputDir = Join-Path $TestDrive "testReportOutput"
         New-Item -Type Directory $InputDir | Out-Null
+        return $InputDir, $OutputDir
+    }
 
-        '
-            <test-results failures="0" inconclusive="0" skipped="0" date="2018-01-01" time="15:00:00">
-                <test-suite name="outer_suite" type="TestFixture" result="Success">
-                    <results>
-                        <test-suite name="inner_suite" type="TestFixture" result="Success">
-                            <results>
-                                <test-case name="test" result="Success" />
-                            </results>
-                        </test-suite>
-                    </results>
-                </test-suite>
-            </test-results>
-        ' | Set-Content -Path (Join-Path $InputDir "foo.xml")
-        
-        Convert-TestReportsToHtml -XmlReportsDir $InputDir -OutputDir $OutputDir
+    function Clear-TemporaryDirs {
+        param([string[]] $Dirs)
+        $Dirs | ForEach-Object {
+            Remove-Item -Recurse -Force $_
+        }
     }
 
-    It "creates appropriate subdirectories" {
-        Join-Path $OutputDir "raw_NUnit" | Should Exist
-        Join-Path $OutputDir "pretty_test_report" | Should Exist
+    function Invoke-FakeReportunit {
```

Comment:
note to reviewers: the reasons I decided to fake report unit are:
* the main reason is that this test suite took 9 seconds to run; while the ideal scenario is all unit tests run under a second. (not possible with Pester but still.)
* also, we don't want to force the developer to install any dependencies (if possible) to develop code
* there *is* a bunch of logic regarding paths, json etc. in this module which needs to be tested, and *can* be tested in separation of reportunit.

I agree that real reportunit exploration/API tests should be somewhere. I believe the answer is: "In CI tests". This is something that will need to be discussed with @katrybacka I guess.